File: analysis/import/SCMLogParser/src/main/java/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/Metric.java
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics;

import de.maibornwolff.codecharta.importer.scmlogparser.input.Commit;
import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification;

public interface Metric {

    String description();

    String metricName();

    Number value();

    default void registerModification(Modification modification) {
        // defaults to: do nothing
    }

    default void registerCommit(Commit commit) {
        // defaults to: do nothing
    }
}

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/Metric.kt
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparser.input.Commit
import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification

interface Metric {

    fun description(): String

    fun metricName(): String

    fun value(): Number

    fun registerModification(modification: Modification) {
        // defaults to: do nothing
    }

    fun registerCommit(commit: Commit) {
        // defaults to: do nothing
    }
}
